examples/cis3-nft-sponsored-txs/src/lib.rs

```diff
@@ -0,0 +1,2083 @@
+    if let Receiver::Contract(address, function) = to {
+        let parameter = OnReceivingCis2Params {
+            token_id,
+            amount,
+            from,
+            data,
+        };
+        host.invoke_contract(&address, &parameter, function.as_entrypoint_name(), Amount::zero())?;
+    }
+
+    Ok(())
+}
+
+/// Execute a list of token transfers, in the order of the list.
+///
+/// Logs a `Transfer` event and invokes a receive hook function for every
+/// transfer in the list.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - Any of the transfers fail to be executed, which could be if:
+///     - The `token_id` does not exist.
+///     - The sender is not the owner of the token, or an operator for this
+///       specific `token_id` and `from` address.
+///     - The token is not owned by the `from`.
+/// - Fails to log event.
+/// - Any of the receive hook function calls rejects.
+#[receive(
+    contract = "cis3_nft",
+    name = "transfer",
+    parameter = "TransferParameter",
+    error = "ContractError",
+    enable_logger,
+    mutable
+)]
+fn contract_transfer<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+    logger: &mut impl HasLogger,
+) -> ContractResult<()> {
+    // Parse the parameter.
+    let TransferParams(transfers): TransferParameter = ctx.parameter_cursor().get()?;
+    // Get the sender who invoked this contract function.
+    let sender = ctx.sender();
+
+    for Transfer {
+        token_id,
+        amount,
+        from,
+        to,
+        data,
+    } in transfers
+    {
+        // Authenticate the sender for this transfer
+        ensure!(
+            from == sender || host.state().is_operator(&sender, &from),
+            ContractError::Unauthorized
+        );
+
+        _transfer(token_id, amount, from, to, data, host, logger)?;
+    }
+    Ok(())
+}
+
+/// The parameter type for the contract function `permitTransfer`.
+#[derive(Debug, Serialize, SchemaType)]
+pub struct PermitTransferParams(#[concordium(size_length = 2)] pub Vec<PermitTransferParam>);
+
+/// Verify an ed25519 signature and allow the transfer of tokens. Execute a list
+/// of token transfers, in the order of the list.
+///
+/// Logs a `Transfer` event and invokes a receive hook function for every
+/// transfer in the list.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - A different nonce is expected.
+/// - The signature was intended for a different contract.
+/// - The signature was intended for a different `entry_point`.
+/// - No public key was registered for the given account.
+/// - The signature can not be validated.
+/// - Any of the transfers fail to be executed, which could be if:
+///     - The `token_id` does not exist.
+///     - The token is not owned by the `from`.
+/// - Fails to log event.
+/// - Any of the receive hook function calls rejects.
```

Comment:
```suggestion
/// - Any of the receive hook function calls reject.
```

Reply:
Because of the *any/anyone* (singular), I feel it should be `rejects`.

**Any** of the receive hook function calls rejects